### PR TITLE
[8.0] [FIX] purchase_requisition_bid_selection - cancel draftbid on generate po

### PR DESCRIPTION
**Pull request by yvaucher**
Fix cancellation of draftpo remaing in state draftpo instead of being cancelled when generating PO.
Plus add test to reproduce

This PR depends on 
- [x] #92

### CODE REVIEW

**Comment by pedrobaeza** [comment]
:+1:

File: purchase_requisition_bid_selection/tests/test_generate_po.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+import openerp.tests.common as common
+from openerp import fields
+
+
+class test_generate_po(common.TransactionCase):
+    """ Test genertation of a purchase order from selected Bid and ensure
```
Comment (lepistone) [inline purchase_requisition_bid_selection/tests/test_generate_po.py]: typo

File: purchase_requisition_bid_selection/tests/test_generate_po.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        # draftbid is at same workflow state as draft
+        self.draftbid = PO.create(
+            {'state': 'draftbid',
+             'partner_id': partner_14.id,
+             'location_id': self.env.ref('stock.stock_location_stock').id,
+             'pricelist_id': partner_14.property_product_pricelist_purchase.id,
+             })
+
+        purchases = self.env['purchase.order']
+        purchases |= self.bid_selected
```
Comment (lepistone) [inline purchase_requisition_bid_selection/tests/test_generate_po.py]: this is perfectly fine for me, but you can also say
`purchases = self.bid_selected | self.bid | self.draftbid`

**Comment by lepistone** [comment]
:+1:, and thanks for the cool workflow stubbing - it's not what we want to test here.

File: purchase_requisition_bid_selection/tests/test_generate_po.py (hunk 1 above)
Comment (yvaucher) [inline purchase_requisition_bid_selection/tests/test_generate_po.py]: s/genertation/generation

**Comment by yvaucher** [comment]
@lepistone I amended fixes in last commit

**Comment by max3903** [comment]
:+1:

**Pull request merged**